src/library/dlgtrackinfo.cpp

```diff
@@ -423,23 +443,25 @@ void DlgTrackInfo::saveTrack() {
         if (!pCue) {
             continue;
         }
-
         updatedRows.insert(oldRow);
 
         QVariant vHotcue = hotcueItem->data(Qt::DisplayRole);
-        if (vHotcue.canConvert(QMetaType::Int)) {
-            int iTableHotcue = vHotcue.toInt();
+        bool ok;
+        int iTableHotcue = vHotcue.toInt(&ok);
+        if (ok) {
             // The GUI shows hotcues as 1-indexed, but they are actually
             // 0-indexed, so subtract 1
             pCue->setHotCue(iTableHotcue - 1);
         } else {
             pCue->setHotCue(-1);
         }
 
-        auto colorComboBox = qobject_cast<QComboBox*>(colorWidget);
-        if (colorComboBox) {
-            PredefinedColorPointer color = Color::kPredefinedColorsSet.allColors.at(colorComboBox->currentIndex());
-            pCue->setColor(color);
+        if (pCue->getType() == Cue::CUE) {
+            auto colorComboBox = qobject_cast<QComboBox*>(colorWidget);
+            if (colorComboBox) {
+                PredefinedColorPointer color = Color::kPredefinedColorsSet.allColors.at(colorComboBox->currentIndex());
```

Comment:
Just for the record: Obtaining a value and passing it along is a typical use case where `auto` is justified. Repeating the actual type if you don't even use it adds nothing.